script.js

```diff
@@ -1,12 +1,138 @@
-//You can edit ALL of the code here
+let container = document.getElementById("root");
+let searchContainer = document.getElementById("search-container");
+let allEpisodes;
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  makeDropdownMenu(allEpisodes);
+  searchForEpisode();
 }
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let rowMainElem = document.createElement("section");
+  rowMainElem.classList.add("row");
+
+  for (let episode of episodeList) {
+    let episodeContainer = document.createElement("div");
+    episodeContainer.classList.add("col-4");
+
+    let title = document.createElement("h4");
+    let seasonEpisodeNumber = document.createElement("h5");
+
+    let imageDiv = document.createElement("div");
+    let image = document.createElement("img");
+
+    let episodeSummary = document.createElement("div");
+
+    title.innerText = `${episode.name}`;
+
+    seasonEpisodeNumber.innerText = makeIdForEpisodes(episode);
+
+    image.src = episode.image.medium;
+    imageDiv.appendChild(image);
+
+    episodeSummary.innerHTML = `${episode.summary}`;
+    episodeContainer.append(title, seasonEpisodeNumber, image, episodeSummary);
+    rowMainElem.appendChild(episodeContainer);
+  }
+  container.appendChild(rowMainElem);
+}
+
+function searchForEpisode() {
+  let searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.placeholder = "Search for Episode";
+
+  searchContainer.appendChild(searchInput);
+  let searchResult = document.createElement("div");
+  searchResult.classList.add("search-result");
+
+  searchInput.addEventListener("keyup", (event) => {
+    let searchString = event.target.value.toLowerCase();
+    const allEpisodes = getAllEpisodes();
```

Comment:
You shouldn't need line 52 since `allEpisodes` is a global variable. Right now you are still using the episode list provided, not the result of your fetch.